src/effector/createDomain.ts

```diff
@@ -15,17 +15,35 @@ import {getGraph, getParent} from './getter'
 import {DOMAIN} from './tag'
 import {launch} from './kernel'
 import {calc} from './step'
+import {flattenConfig} from './config'
+import {assert} from './throw'
 
 export function createDomain(nameOrConfig: any, maybeConfig?: any): Domain {
-  const node = createNode({family: {type: DOMAIN}, regional: true})
+  const config = flattenConfig({
+    or: maybeConfig,
+    and: typeof nameOrConfig === 'string' ? {name: nameOrConfig} : nameOrConfig,
+  }) as any
 
-  const result = {
+  assert(
+    !(config?.domain && config?.parent),
+    'Domain cannot be created with two domains as parent. Please, remove {domain} argument.',
```

Comment:
We cannot throw exception here since it would block usage of `effector-root` and `effector-logger`, since they are relying on imports substitution 🤔 

In the actual code of the app we have:
```ts
const myCoolDomain = createDomain()

const event = createEvent({ domain: myCoolDomain })
```

But after we set up `effector-logger` via babel-plugin, we basically have the equivalent of
```ts
const myCoolDomain = loggerDomain.createDomain()

const event = loggerDomain.createEvent({ domain: myCoolDomain })
```
even though actual code of the app is unchanged

In case of throw it just would not be possible to make `logger` work without rewriting app code

I suggest to choose "parent domain override" behavior:
```ts
const event = d1.createEvent({domain: d2})
```
where `d2` would be parent domain of `event` and not `d1` - it would allow to use `effector-logger` and `effector-root` in the same way it was before, without actual changes in the app code

With `logger` it would work because every domain in the app will also belong to the loggerDomain

Reply:
Makes sense. Implemented in 0d619620c5a0d700961c96bd0dcc0ff0b2cd7e68